### PR TITLE
MM-22768 :  Migrate 'components/more_direct_channels.test.jsx' to TypeScript

### PR DESCRIPTION
**Pull request by AbhinavVihan**
#### Summary
Migrated more_direct_channels.jsx to tsx

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/21039
Fixes  https://mattermost.atlassian.net/browse/MM-22768

#### Related Pull Requests
None
#### Screenshots
NA

#### Release Note

```release-note
Migrated more_direct_channels.jsx to tsx

```

### CODE REVIEW

File: components/more_direct_channels/more_direct_channels.test.tsx [hunk 1]
```diff
@@ -223,35 +220,30 @@ describe('components/MoreDirectChannels', () => {
...
+                ...mockedUser,
                 id: 'deleted_user_2',
-                label: 'deleted_user_id_2',
-                value: 'deleted_user_id_2',
                 delete_at: 1,
             },
             {
+                ...mockedUser,
                 id: 'deleted_user_3',
-                label: 'deleted_user_id_3',
```
Comment (AshishDhama) [inline components/more_direct_channels/more_direct_channels.test.tsx:253]: @AbhinavVihan Why are we removing these

**Review by M-ZubairAhmed — APPROVED** [review]

File: components/more_direct_channels/more_direct_channels.test.tsx (hunk 1 above)
Comment (AbhinavVihan) [reply components/more_direct_channels/more_direct_channels.test.tsx:253]: coz UserProfile does not contain these keys and we don't actually need these here.

**Review by AshishDhama — APPROVED** [review]

**Comment by AshishDhama** [comment]
@jgilliam17 please let me know if need any help in testing this.

**Review by jgilliam17 — APPROVED** [review]
Thanks @AbhinavVihan 
Tested, looks good to merge.
[E2E report](https://automation-dashboard.vercel.app/cycles/466f9ad8-ae5b-4956-b48f-eac137de38d2) - no PR related failures.

**Comment by M-ZubairAhmed** [comment]
@AbhinavVihan we do not put changelogs for ts migrations.

**Pull request merged**